tests/test_scheduler_plan_mode.rs

```diff
@@ -0,0 +1,307 @@
+        .join("tests")
+        .join("minimal_suite");
+
+    let config = create_config(
+        &test_dir,
+        &suite_dir,
+        RCCConfig {
+            binary_path: var("RCC_BINARY_PATH")?.into(),
+            profile_config: RCCProfileConfig::Custom(create_custom_rcc_profile(&test_dir)?),
+            robocorp_home_base: temp_dir_path.join("rc_home_base"),
+        },
+        CondaConfig {
+            micromamba_binary_path: var("MICROMAMBA_BINARY_PATH")?.into(),
+            base_directory: temp_dir_path.join("conda_base"),
+        },
+    );
+
+    run_scheduler(
+        &test_dir,
+        &config,
+        Some("rcc_plan"),
+        false,
+        var("N_SECONDS_RUN_MAX")?.parse::<u64>()?,
+    )
+    .await?;
+
+    let results_dir = plan_results_directory(&results_directory(&config.runtime_directory));
+    assert!(results_dir.is_dir(), "Plan results directory should exist");
+
+    let rcc_plan_result = results_dir.join("rcc_plan.json");
+    assert!(
+        rcc_plan_result.is_file(),
+        "rcc_plan.json result file should exist"
+    );
+
+    let conda_plan_result = results_dir.join("conda_plan.json");
+    assert!(
+        !conda_plan_result.is_file(),
+        "conda_plan.json result file should NOT exist (only rcc_plan was executed)"
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+#[ignore]
+async fn test_plan_mode_no_result() -> AnyhowResult<()> {
+    let test_dir = Utf8PathBuf::from(var("TEST_DIR")?);
+    create_dir_all(&test_dir)?;
+
+    let temp_dir = tempfile::tempdir()?;
+    let temp_dir_path: Utf8PathBuf = Utf8PathBuf::try_from(temp_dir.path().to_path_buf())?;
+
+    let suite_dir = Utf8PathBuf::from(var("CARGO_MANIFEST_DIR")?)
+        .join("tests")
+        .join("minimal_suite");
+
+    let config = create_config(
+        &test_dir,
+        &suite_dir,
+        RCCConfig {
+            binary_path: var("RCC_BINARY_PATH")?.into(),
+            profile_config: RCCProfileConfig::Custom(create_custom_rcc_profile(&test_dir)?),
+            robocorp_home_base: temp_dir_path.join("rc_home_base"),
+        },
+        CondaConfig {
+            micromamba_binary_path: var("MICROMAMBA_BINARY_PATH")?.into(),
+            base_directory: temp_dir_path.join("conda_base"),
+        },
+    );
+
+    run_scheduler(
+        &test_dir,
+        &config,
+        Some("rcc_plan"),
+        true,
+        var("N_SECONDS_RUN_MAX")?.parse::<u64>()?,
+    )
+    .await?;
+
+    let results_dir = plan_results_directory(&results_directory(&config.runtime_directory));
+    let rcc_plan_result = results_dir.join("rcc_plan.json");
+
+    assert!(
+        !rcc_plan_result.is_file(),
+        "rcc_plan.json result file should NOT exist when --no-plan-result is specified"
+    );
+
+    let working_dir = config
+        .runtime_directory
+        .join("working")
+        .join("plans")
+        .join("rcc_plan");
+    assert!(
+        working_dir.is_dir(),
```

Comment:
We can make this assertion even stronger. There should be exactly one subdirectory in the plan working directory (we get one subdirectory per plan execution and we only have a single execution here). And this subdirectory should be non-empty.